### PR TITLE
ci: remove pull_request_target

### PR DESCRIPTION
**Pull request by aaron-steinfeld**
## Description

Merge build and test workflows into same workflow and scheduled them. They had been separated to allow secret access in forks, but that's not really necessary - changed it to omit steps in forks that require secrets.

### CODE REVIEW

**Review by tim-mwangi — APPROVED** [review]

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
+        with:
+          username: ${{ secrets.DOCKERHUB_READ_USER }}
+          password: ${{ secrets.DOCKERHUB_READ_TOKEN }}
+
+      - name: Build with Gradle
+        uses: hypertrace/github-actions/gradle@main
+        with:
+          args: assemble dockerBuildImages
+
+      - name: Run Trivy vulnerability scanner
```
Comment (ravisingal) [inline .github/workflows/build-and-test.yml:32]: should we add `if` condition here as well?

Comment (aaron-steinfeld) [reply .github/workflows/build-and-test.yml:32]: hmm let me go check inside the action - from the outside it appears safe to run on a fork

Comment (aaron-steinfeld) [reply .github/workflows/build-and-test.yml:32]: So the upload would presumably fail - perhaps change output mode for forks to log?

Comment (aaron-steinfeld) [reply .github/workflows/build-and-test.yml:32]: I had changed the local one here to 
```
output-mode: ${{ (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository) && 'github' || 'log' }}
```
but if we can put that defaulting into the action it'd be cleaner.

Comment (aaron-steinfeld) [reply .github/workflows/build-and-test.yml:32]: backed my local change out in favor of the shared action change

**Review by ravisingal — APPROVED** [review]

**Pull request merged**